echo.py

```diff
@@ -0,0 +1,12 @@
+import socket
+
+HOST = '127.0.0.1'  # Listen on all interfaces
+PORT = 12345      # Arbitrary non-privileged port
```

Comment:
Consider using `process.argv[]` for these.  I'd want tests that include IPv6 as well.